File: Ticketing_Portal_Project/Backend/Libraries/TicketingLibrary/Models/TicketReply.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TicketingLibrary.Models
{
    [Table("TicketReply")]
    public class TicketReply
    {
        [Key]
        [Column(TypeName = "CHAR(6)")]
        public string? ReplyId { get; set; }

        [Column(TypeName = "CHAR(4)")]
        public string? TicketId { get; set; }

        [Column(TypeName = "CHAR(4)")]
        public string? ReplyByCreatorEmpId { get; set; }

        [Column(TypeName = "CHAR(4)")]
        public string? ReplyByAssignedEmpId { get; set; }

        [ForeignKey("TicketId")]
        public virtual Ticket? Ticket { get; set; }

        [ForeignKey("ReplyByCreatorEmpId")]
        public virtual Employee? ReplyByCreator { get; set; }
        
        [ForeignKey("ReplyByAssignedEmpId")]
        public virtual Employee? ReplyByAssigned { get; set; } 
    }
}

File: Ticketing_Portal_Project/Frontend/angular-app/src/app/models/TicketReply.ts
export class TicketReply {
    public replyId: string;
    public ticketId: string;
    public replyByCreatorEmpId: string;
    public replyByAssignedEmpId: string;
    public replyMessage: string;


    constructor(replyId: string, ticketId: string, replybycreatorempId: string, replybyassignedempId: string, replyMessage: string) {
        this.replyId = replyId;
        this.ticketId = ticketId;
        this.replyByCreatorEmpId = replybycreatorempId;
        this.replyByAssignedEmpId = replybyassignedempId;
        this.replyMessage = replyMessage;
    }
}
 